src/common/telemetryGenerators.ts

```diff
@@ -59,12 +60,9 @@ export abstract class TelemetryGeneratorBase {
     }
 
     public addError(error: Error): TelemetryGeneratorBase {
-        this.add("error.message" + ++this.errorIndex, error.message, /*isPii*/ true);
         let errorWithErrorCode: IHasErrorCode = <IHasErrorCode>(<Record<string, any>>error);
-        if (errorWithErrorCode.errorCode) {
-            this.add("error.code" + this.errorIndex, errorWithErrorCode.errorCode, /*isPii*/ false);
-        }
-
+        const errorCode = errorWithErrorCode.errorCode || InternalErrorCode.UnknownError;
+        this.add("error.code" + ++this.errorIndex, errorCode, /*isPii*/ false);
```

Comment:
We may use template literals here for better readability
```suggestion
        this.add(`error.code${++this.errorIndex}`, errorCode, /*isPii*/ false);
```